### PR TITLE
PORTAL-1005: Add logout route

### PR DESCRIPTION
**Pull request by ian-sawyer**
- Adds a logout route
- Onclick of Logout section of sidebar
- Maintains ability for logout to be a loading page

### CODE REVIEW

File: src/Dashboard.tsx [hunk 1]
```diff
@@ -24,17 +24,24 @@ function Dashboard() {
   const location = useLocation();
   const navigate = useNavigate();
 
+  const [isSessionRestore, setSessionRestore] = useState(false);
```
Comment (cfarmer-fearless) [inline src/Dashboard.tsx:27]: What's this flag for?

**Comment by cfarmer-fearless** [comment]
Seeing some errors in browser console when logging out and then trying to hit the /dashboard page
The loading of “https://apigw-stg.cdc.gov:8443/auth/oauth/v2/authorize/samslogin?action=display&sessionID=3b888227-a845-49e0-9d4e-6e74e563b551&sessionData=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.ew0KICAgICJzZXNzaW9uIjogew0KICAgICAgICAic2Vzc2lvbklEIjoiM2I4ODgyMjctYTg0NS00OWUwLTlkNGUtNmU3NGU1NjNiNTUxIiwNCiAgICAgICAgImV4cCI6MTY5NDYzNjgzNiwNCiAgICAgICAgImN1cnJlbnRfdXNlcm5hbWUiOiIiLA0KICAgICAgICAiY3VycmVudF91c2VyX2NvbnNlbnQiOiJub25lIiwNCiAgICAgICAgImN1cnJlbnRfdXNlcl9yb2xlIjoiIiwNCiAgICAgICAgImN1cnJlbnRfdXNlcl9hY3IiOiIwIiwNCiAgICAgICAgImN1cnJlbnRfdXNlcl9hdXRoVGltZSI6IjAiLA0KICAgICAgICAic2FsdCI6IiIsDQogICAgICAgICJ0aGlyZF9wYXJ0eV9zc29fdG9rZW4iOiIiLA0KICAgICAgICAidGhpcmRfcGFydHlfc3NvX3Rva2VuX3R5cGUiOiIiDQogICAgfSwNCiAgICAicmVxdWVzdF9jb25zZW50Ijogew0KICAgICAgICAiY2xpZW50X25hbWUiOiJEZXhkZXYiLA0KICAgICAgICAic2NvcGVfdmVyaWZpZWQiOiJvcGVuaWQrcHJvZmlsZStlbWFpbCIsDQogICAgICAgICJyZXNvdXJjZV92ZXJpZmllZCI6Imh0dHBzJTNBJTJGJTJGYXBpZ3ctc3RnLmNkYy5nb3YlMkYqIg0KICAgIH0sDQogICAgInJlcXVlc3RfcGFyYW1ldGVycyI6IHsNCiAgICAgICAgImRpc3BsYXkiOiJwYWdlIiwNCiAgICAgICAgInByb21wdCI6Im5vbmUiLA0KICAgICAgICAiaWRfdG9rZW5faGludCI6IiIsDQogICAgICAgICJsb2dpbl9oaW50IjoiIiwNCiAgICAgICAgImFjcl92YWx1ZXMiOiIiLA0KICAgICAgICAiY2xpZW50X2lkIjoiMDM0MmRlZWQtNzNmZC00ZTQ1LWIzMDYtYmEzY2JlZWJlNjI4IiwNCiAgICAgICAgIm5vbmNlIjoiIiwNCiAgICAgICAgInNjb3BlIjoib3BlbmlkK3Byb2ZpbGUrZW1haWwiLA0KICAgICAgICAicmVzb3VyY2UiOiIiLA0KICAgICAgICAibWF4X2FnZSI6ICIiDQogICAgfQ0KfQ.oY-XwvX8FpUs6TXQL0LoPEerU6thYuvia36LLUiWMh0” in a frame is denied by “X-Frame-Options“ directive set to “deny“.

**Comment by cfarmer-fearless** [comment]
Seeing this error in the SWA CLI when logging out and trying to hit the /dashboard page:
Timed out waiting for: tcp:127.0.0.1:5173

File: src/Dashboard.tsx (hunk 1 above)
Comment (ian-sawyer) [reply src/Dashboard.tsx]: I also was able to remove this hook. Branch updated.

**Comment by ian-sawyer** [comment]
@cfarmer-fearless I was able to resolve the errors. Branch updated.

File: src/Dashboard.tsx [hunk 2]
```diff
@@ -32,9 +33,19 @@ function Dashboard() {
...
+      const oidcStorage = sessionStorage.getItem(
+        `oidc.user:${getEnv("VITE_SAMS_AUTHORITY_URL")}:${getEnv(
+          "VITE_SAMS_CLIENT_ID"
+        )}`
+      );
+
+      if (oidcStorage) {
+        auth.signinSilent();
+      } else {
+        navigate("/", { replace: true });
```
Comment (cfarmer-fearless) [inline src/Dashboard.tsx:45]: The errors in my past comments are resolved, but I am still seeing the "Timed out waiting for tcp:127.0.0.1:5173" warning message.  The OIDC lib must be still trying to sign in somehow.  In addition, I am seeing a flash of the dashboard when I click the browser back button after logging out.  We could solve that by moving this `navigate` call outside of the use effect, and using the `Navigate` component instead.

**Review by cfarmer-fearless — CHANGES_REQUESTED** [review]

File: src/Dashboard.tsx (hunk 2 above)
Comment (ian-sawyer) [reply src/Dashboard.tsx:45]: Ok, updating now.

**Review by cfarmer-fearless — APPROVED** [review]

**Pull request merged**